src/main/java/uk/gov/hmcts/reform/idam/client/IdamApi.java

```diff
@@ -33,9 +35,11 @@ AuthenticateUserResponse authenticateUser(
         @RequestHeader(HttpHeaders.AUTHORIZATION) final String authorisation,
         @RequestParam("response_type") final String responseType,
         @RequestParam("client_id") final String clientId,
-        @RequestParam("redirect_uri") final String redirectUri
+        @RequestParam("redirect_uri") final String redirectUri,
+        @RequestBody String requestBody
```

Comment:
can't you create a class for this?
AuthorizeUserRequest

and shouldn't all the above parameters be removed, response_type etc?

Reply:
Creating an object and passing it as request body doesn't work. Only MultiValue map option works using FormEncoder which is implemented now.
Also removed the request params.